lib/client/api.go

```diff
@@ -457,6 +471,30 @@ func (p *ProfileStatus) DatabaseServices() (result []string) {
 	return result
 }
 
+// DatabasesForCluster returns a list of databases for this profile, for the
+// specified cluster name.
+func (p *ProfileStatus) DatabasesForCluster(clusterName string) ([]tlsca.RouteToDatabase, error) {
+	if clusterName == "" || clusterName == p.Cluster {
+		return p.Databases, nil
+	}
```

Comment:
Is this special case needed? Maybe we can just get rid of `p.Databases` field completely since profile now has a method for returning databases for the specified cluster?

Reply:
I just found that `tsh logut` is using calling `dbprofile.Delete` on each of profile.Database.

I can certainly change this to use the new function. But this is a problem if we have logined db from another cluster, and we are not calling dbprofile.Delete on dbs in other clusters.

Reply:
Added issue #9250 for tracking the dbprofile.Delete problem

I found a few places where `p.Databases` are still used. Mainly when we looping through all profiles.

For example, in tsh logout, we are trying to get databases in each profile
https://github.com/gravitational/teleport/blob/e9382011da5649a0e4c66616abb9e63f51c628e7/tool/tsh/tsh.go#L1131

Similarly, we `printStatus` are all profiles and their databases (default cluster)
https://github.com/gravitational/teleport/blob/e9382011da5649a0e4c66616abb9e63f51c628e7/tool/tsh/tsh.go#L2132

I am debating to keep this special case for these scenarios for now